draft-ietf-mls-protocol.md

```diff
@@ -5880,39 +5910,42 @@ Template:
 
 * Recommended: Same as in {{mls-ciphersuites}}
 
+* External: Whether a proposal of this type may be sent by an `external` sender
+  (see {{external-proposals}}).
+
 * Path Required: Whether a Commit covering a proposal of this type is required
   to have its `path` field populated (see {{commit}}).
 
 * Reference: The document where this extension is defined
 
 Initial contents:
 
-| Value            | Name                     | R | Path | Ref      |
-|:-----------------|:-------------------------|:--|:-----|:---------|
-| 0x0000           | RESERVED                 | - | -    | RFC XXXX |
-| 0x0001           | add                      | Y | N    | RFC XXXX |
-| 0x0002           | update                   | Y | Y    | RFC XXXX |
-| 0x0003           | remove                   | Y | Y    | RFC XXXX |
-| 0x0004           | psk                      | Y | N    | RFC XXXX |
-| 0x0005           | reinit                   | Y | N    | RFC XXXX |
-| 0x0006           | external_init            | Y | Y    | RFC XXXX |
-| 0x0007           | group_context_extensions | Y | Y    | RFC XXXX |
-| 0x0A0A           | GREASE                   | Y | -    | RFC XXXX |
-| 0x1A1A           | GREASE                   | Y | -    | RFC XXXX |
-| 0x2A2A           | GREASE                   | Y | -    | RFC XXXX |
-| 0x3A3A           | GREASE                   | Y | -    | RFC XXXX |
-| 0x4A4A           | GREASE                   | Y | -    | RFC XXXX |
-| 0x5A5A           | GREASE                   | Y | -    | RFC XXXX |
-| 0x6A6A           | GREASE                   | Y | -    | RFC XXXX |
-| 0x7A7A           | GREASE                   | Y | -    | RFC XXXX |
-| 0x8A8A           | GREASE                   | Y | -    | RFC XXXX |
-| 0x9A9A           | GREASE                   | Y | -    | RFC XXXX |
-| 0xAAAA           | GREASE                   | Y | -    | RFC XXXX |
-| 0xBABA           | GREASE                   | Y | -    | RFC XXXX |
-| 0xCACA           | GREASE                   | Y | -    | RFC XXXX |
-| 0xDADA           | GREASE                   | Y | -    | RFC XXXX |
-| 0xEAEA           | GREASE                   | Y | -    | RFC XXXX |
-| 0xF000  - 0xFFFF | Reserved for Private Use | - | -    | RFC XXXX |
+| Value            | Name                     | R | Ext | Path | Ref      |
+|:-----------------|:-------------------------|:--|:----|:-----|:---------|
+| 0x0000           | RESERVED                 | - | -   | -    | RFC XXXX |
+| 0x0001           | add                      | Y | Y   | N    | RFC XXXX |
+| 0x0002           | update                   | Y | N   | Y    | RFC XXXX |
+| 0x0003           | remove                   | Y | Y   | Y    | RFC XXXX |
+| 0x0004           | psk                      | Y | Y   | N    | RFC XXXX |
+| 0x0005           | reinit                   | Y | Y   | N    | RFC XXXX |
+| 0x0006           | external_init            | Y | N   | Y    | RFC XXXX |
```

Comment:
This one is problematic because an external init is technically an external proposal. This is at odds with 12.1.8, but I think the problem is there, not here. In my mind, the new member proposal is not an external proposal and should have its own category. That would also reflect that it has its own sender type.
Tl;DR: Leave this as is but rewrite 12.1.8.

Reply:
ExternalInit doesn't have the `external` SenderType, it should be `new_member_commit` if I'm understanding correctly

Reply:
@Bren2010 is correct.  The concept here is:

* `SenderType::member` can send any proposal
* `SenderType::new_member_proposal` can send Add
* `SenderType::new_member_commit` can only send an External Commit (with proposals inline)
* `SenderType::external` can send only what is specified here